### PR TITLE
Schematic rest api

### PR DESCRIPTION
**Pull request by afwillia**
**DON'T MERGE YET! NEED TO SET UP A SCHEMATIC SERVER**

This PR contains: 
1. Add schematic REST API endpoints for:
    - manifest generation
    - manifest validation
    - manifest submission
2. create R package framework for data curator (not yet an actual package though).
3. Add unit tests with testthat.
4. Create github workflow to set up server and run tests.
5. synapse REST API wrappers for getUserProfile and is_certified.

### CODE REVIEW

File: R/schematic_rest_api.R [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+storage_project_datasets <- function(url="http://localhost:3001/v1/storage/project/datasets",
+                                     syn_master_file_view,
+                                     syn_master_file_name,
+                                     project_id,
+                                     input_token) {
+  
+  req <- httr::GET(url,
+                    #add_headers(Authorization=paste0("Bearer ", pat)),
+                    query=list(
+                      syn_master_file_view=syn_master_file_view,
```
Comment (milen-sage) [inline R/schematic_rest_api.R:101]: These parameters will change. We'll sync up when the REST API calls are finalized.

**Comment by milen-sage** [comment]
@afwillia at this point are the main steps before transitioning to schematic REST API and deploying the other changes in this PR:
- deploying the REST API  service on the web
- more extensive user testing

We have a meeting with Sage IT team tomorrow and that should greenlight the REST API web service.

File: ui.R [hunk 2]
```diff
@@ -91,36 +96,64 @@ ui <- shinydashboardPlus::dashboardPage(
...
+        fluidRow(
+          box(
+            id = "box_pick_asset_view",
+            status = "primary",
+            width = 6,
+            title = "Choose an asset view: ",
+            selectInput(
+              inputId = "dropdown_asset_view",
+              label = "Asset View:",
+              choices = unlist(rjson::fromJSON(Sys.getenv("DCA_SYNAPSE_MASTER_FILEVIEW"), simplify=FALSE))#"Generating..."
```
Comment (afwillia) [inline ui.R:137]: Change this to DCA_ASSET_VIEW for consistency

Comment (afwillia) [reply ui.R:137]: On second thought - keep it consistent with schematic config.yml

**Comment by afwillia** [comment]
Closing this because this branch won't be merged into develop.

**Pull request closed without merging**